src/elements/content-sidebar/activity-feed/annotation-thread/useAnnotationThread.js

```diff
@@ -0,0 +1,241 @@
+import useAnnotationAPI from './useAnnotationAPI';
+import type { Annotation, AnnotationPermission } from '../../../../common/types/annotations';
+import type { BoxItemPermission, User } from '../../../../common/types/core';
+import type { FeedItemStatus, Comment, BoxCommentPermission } from '../../../../common/types/feed';
+import type { ElementOrigin, ElementsXhrError } from '../../../../common/types/api';
+
+const normalizeReplies = (repliesArray?: Array<Comment>): { [string]: Comment } => {
+    if (!repliesArray) {
+        return {};
+    }
+    return repliesArray.reduce((prevValues, reply) => {
+        return {
+            ...prevValues,
+            [reply.id]: reply,
+        };
+    }, {});
+};
+
+const denormalizeReplies = (repliesMap: { [string]: Comment }): Array<Comment> => {
+    return Object.keys(repliesMap).map(key => repliesMap[key]);
+};
+
+type Props = {
+    annotationId: string,
+    api: API,
+    currentUser: User,
+    errorCallback: (
+        error: ElementsXhrError | Error,
+        code: string,
+        contextInfo?: Object,
+        origin?: ElementOrigin,
+    ) => void,
+    eventEmitter: EventEmitter,
+    fileId: string,
+    filePermissions: BoxItemPermission,
+};
+
+type UseAnnotationThread = {
+    annotation?: Annotation,
+    annotationActions: {
+        handleAnnotationDelete: ({ id: string, permissions: AnnotationPermission }) => any,
+        handleAnnotationEdit: (id: string, text: string, permissions: AnnotationPermission) => void,
+        handleAnnotationStatusChange: (id: string, status: FeedItemStatus, permissions: AnnotationPermission) => void,
+    },
+    error?: {
+        message?: MessageDescriptor,
+        title?: MessageDescriptor,
+    },
+    isLoading: boolean,
+    replies: Array<Comment>,
+    repliesActions: {
+        handleReplyCreate: (message: string) => void,
+        handleReplyDelete: ({ id: string, permissions: BoxCommentPermission }) => void,
+        handleReplyEdit: (
+            replyId: string,
+            message: string,
+            status?: FeedItemStatus,
+            permissions: BoxCommentPermission,
+        ) => void,
+    },
+};
+
+const useAnnotationThread = ({
+    annotationId,
+    api,
+    currentUser,
+    fileId,
+    filePermissions,
+    errorCallback,
+}: Props): UseAnnotationThread => {
+    const [annotation, setAnnotation] = React.useState<Annotation | typeof undefined>();
+    const [replies, setReplies] = React.useState<{ [string]: Comment }>({});
+    const [error, setError] = React.useState();
+    const [isLoading, setIsLoading] = React.useState<boolean>(true);
+
+    // handling events from Sidebar
+    const onAnnotationDeleteStart = (id: string) => {
+        if (annotation !== undefined && id === annotationId) {
+            setAnnotation(prevAnnotation => ({ ...prevAnnotation, isPending: true }));
+        }
+    };
+
+    const onAnnotationUpdateStart = (id: string) => {
```

Comment:
This function duplicates `onAnnotationDeleteStart`. I think it could be just a single one named `setAnnotationPending` or sth similar and it can be passed to both hook "Start" props callbacks.